### ISSUE TITLE
net-analyzer/sslyze-5.0.6 fails tests (hang)

### ISSUE THREAD

**Issue by asarubbo** [pentoo/pentoo-overlay#1202]
https://blogs.gentoo.org/ago/2020/07/04/gentoo-tinderbox/

Issue: net-analyzer/sslyze-5.0.5 fails tests (hang).
Discovered on: amd64 (internal ref: pentoo_tinderbox)

NOTE:
This machine uses lto with CFLAGS=-flto -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing

Here is a bit of explanation:

-Werror=lto-type-mismatch:
User to find possible runtime issues in packages. It likely means the package is unsafe to build & use with LTO.
For projects using the same identifier but with different types across different files, they must be fixed to be consistent across the codebase.

-Werror=odr:
Used to find possible runtime issues in packages. These bugs are a problem anyway but may be even worse when combined with LTO. C++ code must comply with the One Definition Rule (ODR) - see https://en.cppreference.com/w/cpp/language/definition#One_Definition_Rule.

-Werror=strict-aliasing:
Used to find possible runtime issues in packages. These bugs are a problem anyway but may be even worse when combined with LTO.

Workarounds:
- If upstream is friendly and still active, file a bug upstream. For emulators, codecs, games, or multimedia packages, it may be worth just applying a workaround instead, as upstreams sometimes aren't receptive to these bugs (VALID FOR ALL).
- Use the new 'filter-lto' from flag-o-matic.eclass as it's likely to be unsafe with LTO (VALID FOR lto-type-mismatch - odr).
- Fix it yourself if interested, of course (VALID FOR ALL).
- Append-flags -fno-strict-aliasing (VALID FOR strict-aliasing).
- Use memcpy() but a union is sometimes suitable too (VALID FOR strict-aliasing).
- -fstrict-aliasing is implied by -O2, so this must be addressed in some form (VALID FOR strict-aliasing).

See also: https://marc.info/?l=gentoo-dev&m=165639574126280&w=2
I had to kill the process.

At the time of writing, github-cli does not support attachments. I'm using gists but they can be deleted at a certain point. If you want to save them, please create attachments on your own.

build log: https://gist.github.com/3c4244a6127944b2e3b59d1d9e962212

**Comment by blshkv** [comment]
no, it doesn't. Can you increase timeout time? It took 7 min to complete on my laptop.
```
SKIPPED [1] tests/test_mozilla_tls_profile/test_mozilla_config_checker.py:68: Server needs to be updated; check https://github.com/chromium/badssl.com/issues/483
=========================== 158 passed, 8 skipped, 95 warnings in 420.87s (0:07:00) ============================
>>> Completed testing net-analyzer/sslyze-5.0.5
```

**Comment by asarubbo** [comment]
> Can you increase timeout time? It took 7 min to complete on my laptop.

Hi,

if you see at the end of the build log there is:

> Tinderbox killed the process after 20946 seconds (05h:49m:06s)

**Comment by asarubbo** [comment]
pentoo_tinderbox has reproduced this issue with version 5.0.6 - Updating summary.

**Comment by blshkv** [comment]
Tested with net-analyzer/sslyze-6.3.1 (currently in the overlay): 170 passed, 9 skipped, 0 failed in 860s (14m 20s). No hang. The slow runtime is expected — the test suite makes real network connections to external TLS servers (smtp.gmail.com, imap.comcast.net, pop.comcast.net, ldap.uchicago.edu, jabber.org, etc.). The hang reported in 5.0.5/5.0.6 appears to have been fixed upstream in a later release.

**Comment by blshkv** [comment]
Update: the test suite includes tests that make real external network connections (smtp.gmail.com, imap.comcast.net, pop.comcast.net, ldap.uchicago.edu, jabber.org, www.google.com, badssl.com, etc.) with no network guard. In a properly network-sandboxed Portage environment these would block indefinitely, which explains the original hang.

Fixed in sslyze-6.3.1.ebuild by overriding `python_test()` to ignore those directories:
- `tests/server_connectivity_tests/` — connects to external TLS/opportunistic-TLS servers
- `tests/test_mozilla_tls_profile/` — connects to google.com, mozilla.com, badssl.com

With the network tests excluded: **141 passed, 4 skipped, 0 failed in 739s (12m 19s)** — all using a local OpenSSL server spun up by the test suite.
